components/camel-google/camel-google-functions/src/main/java/org/apache/camel/component/google/functions/GoogleCloudFunctionsProducer.java

```diff
@@ -0,0 +1,330 @@
+                    Message message = getMessageForResponse(exchange);
+                    message.setBody(response);
+                } catch (ApiException ae) {
+                    LOG.trace("listFunctions command returned the error code {}", ae.getStatusCode());
+                    throw ae;
+                }
+            }
+        } else {
+            ListFunctionsRequest request = ListFunctionsRequest.newBuilder()
+                    .setParent(LocationName.of(getConfiguration().getProject(), getConfiguration().getLocation())
+                            .toString())
+                    .setPageSize(883849137) // TODO check it
+                    .setPageToken("pageToken873572522").build();
+            ListFunctionsPagedResponse pagedListResponse = client.listFunctions(request);
+            List<CloudFunction> response = Lists.newArrayList(pagedListResponse.iterateAll());
+            Message message = getMessageForResponse(exchange);
+            message.setBody(response);
+        }
+    }
+
+    private void getFunction(CloudFunctionsServiceClient client, Exchange exchange) throws InvalidPayloadException {
+        if (getConfiguration().isPojoRequest()) {
+            Object payload = exchange.getIn().getMandatoryBody();
+            if (payload instanceof CloudFunctionName) {
+                CloudFunction response;
+                try {
+                    response = client.getFunction((CloudFunctionName) payload);
+                } catch (ApiException ae) {
+                    LOG.trace("getFunction command returned the error code {}", ae.getStatusCode());
+                    throw ae;
+                }
+                Message message = getMessageForResponse(exchange);
+                message.setBody(response);
+            }
+        } else {
+            CloudFunctionName cfName = CloudFunctionName.of(getConfiguration().getProject(),
+                    getConfiguration().getLocation(), getConfiguration().getFunctionName());
+            CloudFunction response = client.getFunction(cfName);
+            Message message = getMessageForResponse(exchange);
+            message.setBody(response);
+        }
+    }
+
+    private void callFunction(CloudFunctionsServiceClient client, Exchange exchange) throws InvalidPayloadException {
+        if (getConfiguration().isPojoRequest()) {
+            Object payload = exchange.getIn().getMandatoryBody();
+            if (payload instanceof CallFunctionRequest) {
+                CallFunctionResponse response;
+                try {
+                    response = client.callFunction((CallFunctionRequest) payload);
+                    Message message = getMessageForResponse(exchange);
+                    message.setHeader(GoogleCloudFunctionsConstants.RESPONSE_OBJECT, response);
+                    message.setBody(response.getResult());
+                } catch (ApiException ae) {
```

Comment:
These exceptions that you log in every function, also the same function will be thrown in other duplicated code (e.g: https://github.com/apache/camel/pull/5205/files#diff-0cab0789896e74525b9f46a8445b841a9a8b2f108b6300c7a5edd970d9c47550R166), you don't need to log that?